src/wfdlgs2.c

```diff
@@ -729,16 +766,39 @@ JAPANEND
                goto Error;
             }
 
-            pCopyInfo->dwFunc =  dwSuperDlgMode-IDM_MOVE+1;
+            // Map IDM_* to FUNC_*
+            switch (dwSuperDlgMode) {
+            case IDM_MOVE:
+               pCopyInfo->dwFunc = FUNC_MOVE;
+               break;
+            case IDM_COPY:
+               pCopyInfo->dwFunc = FUNC_COPY;
+               break;
+            case IDM_DELETE:
+               pCopyInfo->dwFunc = FUNC_DELETE;
+               break;
+            case IDM_RENAME:
+               pCopyInfo->dwFunc = FUNC_RENAME;
+               break;
+            case IDM_SYMLINK:
+               pCopyInfo->dwFunc = FUNC_LINK;
+               break;
+            case IDM_HARDLINK:
+               // If source is a directory then create a junction
+               if (IsDirectory(pCopyInfo->pFrom))
+                  pCopyInfo->dwFunc = FUNC_JUNC;
+               else
+                  pCopyInfo->dwFunc = FUNC_HARD;
+               break;

```

Comment:
In the drag drop code the three cases are distinguished (Symbol, Hard, Junction).  Here the choice between the later two is determined by whether the source is a directory.  Why the difference?  Should we do the same in the drag drop code?

Reply:
The 'drag/drop code' already provides all info needed:
- operation type (FUNC_xxxx)
- object type (file or dir)

The Hardlink menu only provides
- object type (file or dir)

To properly perform the operation in WFMoveCopyDriverThread() we need both 'operation type' and 'object type'.
That's why we deduce the operation type here by determining if it is a dir or not, because junctions can only be created from dirs and hardlinks can only be created files.

In other words: 'drag/drop code' provides everything needed, the 'menu code' not.

One other way would have been ( I tinkered about that a long time )
- Have an extra menu for Junctions, so that we end up here with e.g. IDM_JUNCTON. Then we would have the info needed for WFMoveCopyDriverThread() without 'if'

But I didn't want to clutter the menu with an extra entry for Junction, so i 'hid' junction creation behind the 'Hardlink' menu entry.

Well we could improve the situation by having an improved text on the menu: 'Hardlink/Junction\tShift+F11'
But I decided not to do because this also raises the question with the user: 'Why is Hardlink and Junction almost the same?'

So I decided to go the easy way and 'forgot' the 'Junction' on the menu.

But yes, you spotted a grey area and there are arguments for one side or the other.